src/docs/scripts/extensions.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from typing import List, Optional, Type
+
+import griffe
+
+from django_components.util.misc import get_import_path
+from docs.scripts.mkdocs import get_mkdocstrings_plugin_handler_options, import_object, load_config
+
+SOURCE_CODE_GIT_BRANCH = "master"
+
+
+mkdocstrings_config = get_mkdocstrings_plugin_handler_options() or {}
+is_skip_docstring: bool = mkdocstrings_config.get("show_if_no_docstring", "false") != "true"
+
+
+# Replacement for `show_bases: true`. Our implementation takes base names from the runtime
+# objects, effectively resolving aliases / reassignments.
+class RuntimeBasesExtension(griffe.Extension):
```

Comment:
These are extensions for `mkdocstrings`. I don't have a very deep grasp of how they work, I got help from the maintainer of mkdocstrings about how to write the plugins.

One important thing to keep in mind is that `mkdocstrings` / `griffe` goes through the source code and parses the AST to discover what's defined, and use the docstrings for documentation.

That's also the reason why I added this `RuntimeBasesExtension`. Since `mkdocstrings` by default uses only the AST, then if there is a class that subclasses another:

```py
class Dog(Animal):
    ...
```

Then `mkdocstrings` is unable to resolve the reference class `Animal`.

So if the original file actually contained this:
```py
Animal = Rock

class Dog(Animal):
    ...
```

Then the documentation would still show `Animal` as the base for `Dog`, although `Animal` is just an alias for `Rock`.

And instead, this custom `RuntimeBasesExtension` imports the class as a Python object to get the actual name of the base classes, and show those.

---

Note: I might copy the code for the documentation for other projects like the component library, that's why I took the time to get this right. Because there, I use aliases so that the end-user could either just the component type or the component class:

```py
class ADividerProps(TypedDict):
    abc: int

class ADividerSlots(TypedDict):
    default: NotRequired[SlotContent]

ADividerType = Component[EmptyTuple, ADividerProps, ADividerSlots, Any, Any, Any]

class ADivider(ADividerType):
    ...
```

And there, using `show_bases: true` would show `ADividerType` as the base class. But this custom extension would successfully resolve that to `Component`.